src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -609,36 +669,143 @@ public void calculateContingencySensitivityValues(PropagatedContingency continge
                 resultWriter.writeContingencyStatus(contingency.getIndex(), SensitivityAnalysisResult.Status.NO_IMPACT);
             }
 
-            DenseMatrix newFlowStates = setReferenceActivePowerFlows(lfNetwork, lfParameters, equationSystem, j, factors,
+            DenseMatrix newFlowStates = calculateActivePowerFlows(lfNetwork, lfParameters, equationSystem, j, factors,
                     newParticipatingElements, disabledBuses, disabledBranches, reporter);
 
             calculateSensitivityValues(factors, newFactorStates, contingenciesStates, newFlowStates, contingencyElements, contingency, resultWriter);
 
             networkState.restore();
-            if (participatingElementsChanged) {
+            if (participatingElementsChanged || rhsChanged) {
```

Comment:
@Hadrien-Godard can you please validate this little fix please?

Reply:
Yes I validate it